### PR TITLE
[ISSUE-339] Optimize retry logic in send shuffle data

### PR DESCRIPTION
**Pull request by xianjingfeng**
### What changes were proposed in this pull request?

1. Set the default value of `rss.client.retry.max` to 50
2. Set  `rss.client.retry.max`  to  `rss.client.retry.max/2` if  replica > replicaWrite
3. Throw an exception if `rss.client.retry.max * rss.client.retry.interval.max > rss.client.send.check.timeout.ms`
### Why are the changes needed?

More reasonable. #339 

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

Added

### CODE REVIEW

File: client/src/main/java/org/apache/uniffle/client/factory/ShuffleClientFactory.java [hunk 1]
```diff
@@ -49,6 +49,11 @@ public ShuffleWriteClient createShuffleWriteClient(
       String clientType, int retryMax, long retryIntervalMax, int heartBeatThreadNum,
       int replica, int replicaWrite, int replicaRead, boolean replicaSkipEnabled, int dataTransferPoolSize,
       int dataCommitPoolSize, int unregisterThreadPoolSize, int unregisterRequestTimeoutSec) {
+    // If replica > replicaWrite, blocks maybe will be sended for 2 rounds.
+    // We need retry less times in this case for let the first round fail fast.
+    if (replica > replicaWrite) {
```
Comment (jerqi) [inline client/src/main/java/org/apache/uniffle/client/factory/ShuffleClientFactory.java:54]: Should we use `replicaSkipEnabled`?

Comment (xianjingfeng) [reply client/src/main/java/org/apache/uniffle/client/factory/ShuffleClientFactory.java]: Done

**Review by jerqi — APPROVED** [review]
LGTM, thanks @xianjingfeng , wait for CI

**Comment by jerqi** [comment]
Merged.

**Pull request merged**